hudi-common/src/main/java/org/apache/hudi/common/table/log/HoodieMergedLogRecordScanner.java

```diff
@@ -282,6 +285,13 @@ protected void processNextDeletedRecord(DeleteRecord deleteRecord) {
 
       Comparable curOrderingVal = oldRecord.getOrderingValue(this.readerSchema, this.hoodieTableMetaClient.getTableConfig().getProps());
       Comparable deleteOrderingVal = deleteRecord.getOrderingValue();
+      if (curOrderingVal instanceof Utf8){
```

Comment:
The payload classes are deprecated now (https://hudi.apache.org/releases/release-1.1.0#deprecation-of-hoodierecordpayload) and the old log record scanner is going to be removed.  The new reader path with merge modes can handle the event time-based merging correctly.  @cbg-wx could you check?

Reply:
@yihua this is a PR for 0.x-branch, should be a critical fix though.